### DISCUSSION TITLE
Map specific fields of child object

### DISCUSSION THREAD

**Discussion by lubumbax** [mapstruct/mapstruct#3339] [Q&A] [answered]
I am using MapStruct 1.5.5.Final (and Java 1.8).
I am trying to map only specific fields of a child object from the parent mapper by specifying those fields with `@Mapping`:

```java
public class Parent {
    Integer id;
    String name;
    Child child;
}

public class Child {
    Integer id;
    String name;
}

// The DTO's are identical

@Mapper
public interface ParentMapper {
    ParentMapper INSTANCE = Mappers.getMapper(ParentMapper.class);

    @BeanMapping(ignoreByDefault = true)
    @Mapping(target = "id", source = "id")
    @Mapping(target = "name", source = "name")
    @Mapping(target = "child.name", source = "child.name")
    ParentDto convert(Parent parent);
}
```

The generated `ParentMapperImpl` results in the following two methods:
```java
public ParentDto convert(Parent parent) {
    if (parent == null) {
        return null;
    } else {
        ParentDto.ParentDtoBuilder parentDto = ParentDto.builder();
        parentDto.child(this.childToChildDto(parent.getChild()));
        parentDto.id(parent.getId());
        parentDto.name(parent.getName());
        return parentDto.build();
    }
}

protected ChildDto childToChildDto(Child child) {
    if (child == null) {
        return null;
    } else {
        ChildDto.ChildDtoBuilder childDto = ChildDto.builder();
        childDto.name(child.getName());
        childDto.id(child.getId());                  // <== Why is this one here?
        return childDto.build();
    }
}
```

Which means, **all the fields from `Child` are being mapped to `ChildDto`**. 
Is that the expected behaviour? 
If it is, what's the best way to map just one field from `Child` into `ChildDto`?
If I explicitly ignore the rest of the fields from `ChildDto` (eg `@Mapping(target = "child.id", ignore = true)`) then it works as I want, but, in case that the child had a lot of fields, I would expect that the `@BeanMapping(ignoreByDefault = true)` would have applied implicitly to the children as well.

This might be related to #3158 but I am not sure.

**Comment by filiphr** [comment, marked as answer]
@lubumbax @lubumbax this is indeed related to #3158. Your example works properly on main.

btw a small note for `@Mapping(target = "name", source = "name")` when the target and source are identical you can just write `    @Mapping(target = "name")`
